internal/stream/concat.go

```diff
@@ -0,0 +1,37 @@
+package stream
+
+import (
+	"github.com/genjidb/genji/internal/expr"
+	"github.com/genjidb/genji/internal/stringutil"
+)
+
+// A ConcatOperator concatenates two streams.
+type ConcatOperator struct {
+	baseOperator
+	S1 *Stream
+	S2 *Stream
+}
+
+// Concat turns two individual streams into one.
+func Concat(s1 *Stream, s2 *Stream) *ConcatOperator {
+	return &ConcatOperator{S1: s1, S2: s2}
+}
+
+func (op *ConcatOperator) Iterate(in *expr.Environment, fn func(*expr.Environment) error) error {
+	err := op.S1.Iterate(in, func(out *expr.Environment) error {
+		fn(out)
```

Comment:
We don't check for errors returned by fn here, is there a reason?

I think this should be `op.S1.Iterate(in, fn)`

Same below

Reply:
Nope, just me not having the linters enabled, my bad.